ibtk/src/math/HierarchyMathOps.cpp

```diff
@@ -2053,20 +2062,40 @@ HierarchyMathOps::interp(const int dst_idx,
 
             d_patch_math_ops.interp(dst_data, src_data, patch);
 
-            if ((ln > d_coarsest_ln) && src_cf_bdry_synch)
+            if (ln > d_coarsest_ln)
             {
-                Pointer<OutersideData<NDIM, double> > os_data = patch->getPatchData(d_os_idx);
-                os_data->copy(*src_data);
+                if (src_cf_bdry_synch)
+                {
+                    Pointer<OutersideData<NDIM, double> > os_data = patch->getPatchData(d_os_idx);
+                    os_data->copy(*src_data);
+                }
+                if (dst_cf_bdry_synch)
+                {
+                    Pointer<OuternodeData<NDIM, double> > on_data = patch->getPatchData(d_on_idx);
```

Comment:
We might need to be careful here --- the code that I wrote to set up `d_on_idx` made it scalar in 2D and vector-valued in 3D (since I was using it to synch up the vorticity).

Reply:
It might simplify thing to use helper functions like `getOuternodePatchDataIndex(depth)` that will manage setting up / tracking the patch data indices and variables for the different depths instead of doing this "by hand".

Reply:
Yup - that's where the failing test fails. It also fails because SAMRAI can generate patches which whose sizes are not integer multiples of the refinement ratio - good thing we added all these tests.